### PR TITLE
Multitask training loss reporting temp fix  + better multitasks task names from task IDs

### PR DESCRIPTION
**Pull request by jaseweston**

### CODE REVIEW

File: parlai/core/worlds.py [hunk 1]
```diff
@@ -249,7 +249,9 @@ def __init__(self, opt, agents, shared=None):
             # Add passed in agents directly.
             self.agents = agents
         self.acts = [None] * len(self.agents)
-
+        if type(self.agents) == list and len(self.agents) > 0:
```
[the remark this answers is not in the document]
Comment (jaseweston) [reply parlai/core/worlds.py]: i think it can be None,..

[the remark this answers is not in the document]
Comment (stephenroller) [reply parlai/core/worlds.py]: then the better check is `if self.agents is not None`

**Comment by jaseweston** [comment]
ok, can u look again plz

**Comment by jaseweston** [comment]
ok no lint errors

File: parlai/scripts/train_model.py [hunk 2]
```diff
@@ -426,11 +512,27 @@ def _average_dicts(self, all_versions):
...
+            metrics[
+                'warning'
+            ] = 'agent level metrics (e.g. loss, mean_loss, ppl) are averaged over tasks'
+
     def _sync_training_metrics(self, metrics):
         """
         Sync training metrics across workers. A handful of special cases are handled
         as exceptions, and the remaining metrics are simply averaged across workers.
         """
+        self._cleanup_inaccurate_metrics(metrics)
```
Comment (stephenroller) [inline parlai/scripts/train_model.py]: Only remaining change is to make this not called from _sync_training_metrics, and call it in the log() function instead.

Comment (jaseweston) [reply parlai/scripts/train_model.py]: moved

**Comment by stephenroller** [comment]
Awesome, thanks for moving that call.

**Pull request merged**